core/src/worker/data-feed.ts

```diff
@@ -335,7 +337,13 @@ function heartbeatJob(aggregatorQueue: Queue, state: State, _logger: Logger) {
       }
       logger.debug(outData, 'outData')
 
-      if (eligibleToSubmit) {
+      if (!eligibleToSubmit) {
+        const msg = `Non-eligible to submit for oracle ${oracleAddress} with operator ${operatorAddress}`
+        throw new OraklError(OraklErrorCode.NonEligibleToSubmit, msg)
+      } else if (!(await isRoundIdFresh(aggregatorQueue, oracleAddress, roundId))) {
+        const msg = `low roundId to submit for oracle ${oracleAddress} with roundId: ${roundId}`
+        throw new OraklError(OraklErrorCode.NonEligibleToSubmit, msg)
```

Comment:
I think we should start reconsidering what do we do with jobs in certain unexpected situations. If we raise an error from inside the job the jobs is moved to failed state and retried if settings allow it. In some cases like this one we do no want to retry the job anymore. We can either just log the error and. return from the function or use `UnrecoverableError` (https://docs.bullmq.io/patterns/stop-retrying-jobs). The example above with submission eligibility is also a bit controversial, I think we should not retry it.